### PR TITLE
Add ability to read cgroup metrics and limits.

### PR DESCRIPTION
**Pull request by andrewkroh**
This gives the ability to ask gosigar for cgroup stats by PID. It will return metrics and limits from the blkio, cpu, cpuacct, and memory subsystems.

How it Works

1. Find the cgroup subsystems supported by the kernel by reading from `/proc/cgroups`.
2. Find the mountpoint for each of the supported cgroups by reading `/proc/self/mountinfo`.
3. For a given PID, find the cgroups that the process is associated with by reading `/proc/[pid]/cgroup`. This file contains an entry for each subsystem and a relative path to the cgroup mount point where metrics and be read.
4. For each subsystem, join the subsystem's mount point with the relative path to the cgroup and read metrics.

### CODE REVIEW

**Comment by ruflin** [comment]
Could you share some details on how you generated the test data?

**Comment by tsg** [comment]
@andrewkroh would it make sense to have that test data committed as a single zip file? I think it's pretty pointless for it show up in diffs, right?

**Comment by andrewkroh** [comment]
@tsg I like the idea of having the test data in a zip file (or hidden in any way). Any suggestions of how to integrate the zipped data with my go tests?

@ruflin The data was copied from the filesystem of my docker-machine. I just started an arbitrary container running bash and ran a few commands to consume some cpu cycles and memory.

**Comment by ruflin** [comment]
@andrewkroh Good to know. Was curious in case we want to update the files in the future.

**Comment by andrewkroh** [comment]
It's easy to recreate if you don't mind updating tests. In the future I'll probably add some test data collected from some other OSes and other tools that use cgroups like systemd and rocket.

**Comment by tsg** [comment]
I think you could do the unzip in a [testing Main](https://golang.org/pkg/testing/#hdr-Main) function. You can also do it only if it folders don't exist, so repeated local runs are fast.

**Comment by andrewkroh** [comment]
I added support for reading the block I/O subsystem metrics. And I moved the test data into a zip file.

**Comment by andrewkroh** [comment]
Later in a separate pull request I want to add logging and some better error messages.

**Comment by andrewkroh** [comment]
Test CLA check

File: cgroup/util.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+func parseUintFromFile(path ...string) (uint64, error) {
+	value, err := ioutil.ReadFile(filepath.Join(path...))
+	if err != nil {
+		return 0, err
+	}
+
+	return parseUint(value)
+}
+
+// parseUint reads a single uint value. It will trip any whitespace before
```
Comment (tsg) [inline cgroup/util.go]: s/trip/trim

**Comment by tsg** [comment]
LGTM, nice work.

**Comment by monicasarbu** [comment]
LGTM, nice code!

**Pull request merged**